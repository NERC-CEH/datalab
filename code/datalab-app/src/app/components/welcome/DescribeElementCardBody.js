import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from 'material-ui/styles';
import { CardContent, CardMedia } from 'material-ui/Card';
import Typography from 'material-ui/Typography';

const bodyText = {
  fontSize: 'larger',
  fontWeight: 'lighter',
};

const styles = theme => ({
  bodyText,
  quoteBodyText: {
    ...bodyText,
    fontStyle: 'italic',
  },
  cardImage: {
    height: 90,
    marginBottom: 8,
  },
});

function DescribeElementCardBody({ classes, content, media, quote }) {
  const generateTextBody = textContent => (
    <CardContent>
      <Typography className={quote ? classes.quoteBodyText : classes.bodyText} type="body1">
        {textContent}
      </Typography>
    </CardContent>
  );

  const generateMediaBody = mediaContent => (
    <CardMedia key="card-body" className={classes.cardImage} image={mediaContent} />
  );

  return media ? generateMediaBody(content) : generateTextBody(content);
}

DescribeElementCardBody.propTypes = {
  classes: PropTypes.object.isRequired,
  content: PropTypes.string.isRequired,
  media: PropTypes.bool,
  quote: PropTypes.bool,
};

export default withStyles(styles)(DescribeElementCardBody);
