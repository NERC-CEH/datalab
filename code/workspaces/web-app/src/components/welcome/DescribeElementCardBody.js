import React from 'react';
import PropTypes from 'prop-types';
import withStyles from '@mui/styles/withStyles';
import CardMedia from '@mui/material/CardMedia';
import Typography from '@mui/material/Typography';
import theme from '../../theme';

const cardContent = {
  height: '100%',
};

const bodyText = {
  ...cardContent,
  fontSize: 'larger',
  fontWeight: 'lighter',
  color: theme.typography.color,
};

const quoteStyle = {
  quotes: '"\\2018" "\\2019"', // left and right single quotations
  fontFamily: 'serif',
  fontSize: '4em',
  position: 'relative',
  top: '0.45em',
  lineHeight: 0,
  letterSpacing: '-0.08em',
  color: 'hsl(0, 0%, 70%)',
};

const styles = injectedTheme => ({
  cardContent,
  bodyText,
  quoteBodyText: {
    ...bodyText,
    paddingTop: injectedTheme.spacing(2),
    paddingLeft: '2em',
    paddingRight: '2em',
    '&::before': {
      ...quoteStyle,
      content: 'open-quote open-quote',
      left: '-0.55em',
      display: 'block',
    },
    '&::after': {
      ...quoteStyle,
      content: 'close-quote close-quote',
      left: '0.05em',
    },
  },
  cardImage: {
    height: 90,
    marginBottom: 8,
  },
});

function DescribeElementCardBody({ classes, content, media, quote }) {
  const generateTextBody = textContent => (
    <div className={classes.cardContent}>
      <Typography className={quote ? classes.quoteBodyText : classes.bodyText} variant="body2">
        {textContent}
      </Typography>
    </div>
  );

  const generateMediaBody = mediaContent => (
    <CardMedia key="card-body" className={classes.cardImage} image={mediaContent} />
  );

  return media ? generateMediaBody(content) : generateTextBody(content);
}

DescribeElementCardBody.propTypes = {
  classes: PropTypes.object.isRequired,
  content: PropTypes.string.isRequired,
  media: PropTypes.bool,
  quote: PropTypes.bool,
};

export default withStyles(styles)(DescribeElementCardBody);
