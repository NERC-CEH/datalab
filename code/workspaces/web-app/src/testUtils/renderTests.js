import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider as MuiThemeProvider } from '@mui/material/styles';
import theme from '../theme';

const AllTheProviders = ({ children }) => (
    <MuiThemeProvider theme={theme}>
        {children}
    </MuiThemeProvider>
);

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
