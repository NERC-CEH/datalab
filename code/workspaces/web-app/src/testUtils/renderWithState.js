import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { render } from '@testing-library/react';
import promise from 'redux-promise-middleware';
import {
  PROMISE_TYPE_PENDING,
  PROMISE_TYPE_SUCCESS,
  PROMISE_TYPE_FAILURE,
} from '../actions/actionTypes';

const promiseTypeSuffixes = [PROMISE_TYPE_PENDING, PROMISE_TYPE_SUCCESS, PROMISE_TYPE_FAILURE];

const TEST_USER = 'test-user';

export const buildDefaultTestState = () => ({
  authentication: {
    identity: {
      sub: TEST_USER,
    },
    permissions: {
      fetching: { inProgress: false },
      value: [],
    },
  },
  projectUsers: {
    fetching: { inProgress: false },
    value: [
      { userId: TEST_USER, role: 'admin' },
    ],
  },
});

export const renderWithState = (
  initialState,
  ComponentToRender,
  props = {},
  renderOptions = {},
) => {
  const mockStore = configureStore([promise({ promiseTypeSuffixes })]);
  const store = mockStore(initialState);

  const wrapper = render(
    <Provider store={store}>
      <ComponentToRender {...props} />
    </Provider>,
    renderOptions,
  );

  return {
    wrapper,
    store,
  };
};
