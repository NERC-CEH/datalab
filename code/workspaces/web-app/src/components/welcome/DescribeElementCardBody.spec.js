import React from 'react';
import { render } from '../../testUtils/renderTests';
import DescribeElementCardBody from './DescribeElementCardBody';

describe('DescribeElementCardBody', () => {
  it('renders correct snapshot', () => {
    const props = {
      content: 'expectedContent',
    };

    expect(render(<DescribeElementCardBody {...props} />).container).toMatchSnapshot();
  });

  it('renders correct snapshot with switches', () => {
    const props = {
      content: 'moreContent',
      quote: true,
      media: true,
    };

    expect(render(<DescribeElementCardBody {...props} />).container).toMatchSnapshot();
  });
});
