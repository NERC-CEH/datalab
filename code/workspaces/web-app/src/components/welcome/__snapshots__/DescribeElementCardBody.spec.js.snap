// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DescribeElementCardBody renders correct snapshot 1`] = `
<div
  className="DescribeElementCardBody-cardContent-1"
>
  <WithStyles(ForwardRef(Typography))
    className="DescribeElementCardBody-bodyText-2"
    variant="body2"
  >
    expectedContent
  </WithStyles(ForwardRef(Typography))>
</div>
`;

exports[`DescribeElementCardBody renders correct snapshot with switches 1`] = `
<WithStyles(ForwardRef(CardMedia))
  className="DescribeElementCardBody-cardImage-4"
  image="moreContent"
  key="card-body"
/>
`;
